tests/autologging/fixtures.py

```diff
@@ -31,3 +49,30 @@ def test_mode_on():
             os.environ[_AUTOLOGGING_TEST_MODE_ENV_VAR] = prev_env_var_value
         else:
             del os.environ[_AUTOLOGGING_TEST_MODE_ENV_VAR]
+
+
+@pytest.fixture(autouse=True)
+def reset_stderr():
```

Comment:
Since some of our tests reassign `stderr` to a test stream to ensure that no content is written during silent mode, this fixture is used to reset `stderr` at the end of each test.